src/ZoomLoop.Api/Features/Vehicles/VehicleIngestHandler.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Quinntyne Brown. All Rights Reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using ZoomLoop.Core;
+using ZoomLoop.Core.Models;
+using ZoomLoop.Core.Services.VehicleIngestion;
+
+namespace ZoomLoop.Api.Features.Vehicles;
+
+public class VehicleIngestHandler : IRequestHandler<VehicleIngestRequest, VehicleIngestResponse>
+{
+    private readonly IZoomLoopContext _context;
+    private readonly IVehicleIngestionService _vehicleIngestionService;
+
+    public VehicleIngestHandler(
+        IZoomLoopContext context,
+        IVehicleIngestionService vehicleIngestionService)
+    {
+        _context = context;
+        _vehicleIngestionService = vehicleIngestionService;
+    }
+
+    public async Task<VehicleIngestResponse> Handle(VehicleIngestRequest request, CancellationToken cancellationToken)
+    {
+        // Convert uploaded images to byte arrays
+        var imageBytes = await ConvertImagesToByteArraysAsync(request.Images, cancellationToken);
+
+        // Call the ingestion service with the images
+        var ingestionRequest = new VehicleIngestionRequest
+        {
+            Images = imageBytes.ToArray()
+        };
+
+        var ingestionResult = await _vehicleIngestionService.IngestVehicleAsync(ingestionRequest, cancellationToken);
+
+        // Find or create Make
+        var make = await _context.Makes
+            .FirstOrDefaultAsync(m => m.Name == ingestionResult.Make, cancellationToken);
+
+        if (make == null)
+        {
+            make = new Make
+            {
+                MakeId = Guid.NewGuid(),
+                Name = ingestionResult.Make,
+                LogoUrl = string.Empty,
+                IsActive = true,
+                DisplayOrder = 0
+            };
+            _context.Makes.Add(make);
+        }
+
+        // Find or create VehicleModel
+        var vehicleModel = await _context.VehicleModels
+            .FirstOrDefaultAsync(vm => vm.MakeId == make.MakeId && vm.Name == ingestionResult.Model, cancellationToken);
```

Comment:
Change query to first Or Default when the VIN matches. Update any Models to include a VIN property of that is missing. Create the necessary data migrations and update the tests